jbpm-event-emitters/jbpm-event-emitters-kafka/src/main/java/org/jbpm/event/emitters/kafka/KafkaEventEmitter.java

```diff
@@ -110,55 +112,40 @@ public void deliver(Collection<InstanceView<?>> data) {
                 producer.send(new ProducerRecord<>(getTopic(topic), mapper.writeValueAsBytes(
```

Comment:
you need to add a callback here to know if the message was delivered or not.

https://kafka.apache.org/10/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html#send-org.apache.kafka.clients.producer.ProducerRecord-org.apache.kafka.clients.producer.Callback-

in case the callback is a exception the data needs to be logged entirely

Reply:
Thanks for pointing it out, done